cmake/s1.cmake

```diff
@@ -11,64 +11,124 @@ if (CONFIG_MCUBOOT AND CONFIG_MCUBOOT_BUILD_S1_VARIANT)
 # most of the variables required for compiling/linking is only available
 # in the scope of this file.
 
-  # Create linker script which has an offset from S0 to S1.
+  set(link_variant_name s1)
+  set(link_variant ${link_variant_name}_)
+
+  configure_linker_script(
+    linker_${link_variant}prebuilt.cmd
+    "-DLINK_INTO_${link_variant_name}"
+    ${PRIV_STACK_DEP}
+    ${CODE_RELOCATION_DEP}
+    ${OFFSETS_H_TARGET}
+    )
+
+  add_custom_target(
+    linker_${link_variant}prebuilt_target
+    DEPENDS
+    linker_${link_variant}prebuilt.cmd
+    )
+
+  # Create prebuilt executable, which is used as input for ISR generation.
+  set(${link_variant}prebuilt ${link_variant}image_prebuilt)
+  add_executable(${${link_variant}prebuilt} ${ZEPHYR_BASE}/misc/empty_file.c)
+  toolchain_ld_link_elf(
+    TARGET_ELF            ${${link_variant}prebuilt}
+    OUTPUT_MAP            ${PROJECT_BINARY_DIR}/${${link_variant}prebuilt}.map
+    LIBRARIES_PRE_SCRIPT  ""
+    LINKER_SCRIPT         ${PROJECT_BINARY_DIR}/linker_${link_variant}prebuilt.cmd
+    LIBRARIES_POST_SCRIPT ""
+    DEPENDENCIES          ""
+    )
+  set_property(TARGET ${${link_variant}prebuilt} PROPERTY LINK_DEPENDS ${PROJECT_BINARY_DIR}/linker_${link_variant}prebuilt.cmd)
+  add_dependencies(   ${${link_variant}prebuilt} linker_${link_variant}prebuilt_target ${PRIV_STACK_DEP} ${LINKER_SCRIPT_TARGET} ${OFFSETS_LIB})
+
+  if(CONFIG_GEN_ISR_TABLES)
+    # ${link_variant}isr_tables.c is generated from ${ZEPHYR_PREBUILT_EXECUTABLE} by
+    # gen_isr_tables.py
+    set(obj_copy_cmd "")
+    bintools_objcopy(
+      RESULT_CMD_LIST obj_copy_cmd
+      TARGET_INPUT    ${OUTPUT_FORMAT}
+      TARGET_OUTPUT   "binary"
+      SECTION_ONLY    ".intList"
+      FILE_INPUT      $<TARGET_FILE:${${link_variant}prebuilt}>
+      FILE_OUTPUT     "${link_variant}isrList.bin"
+      )
+
+    add_custom_command(
+      OUTPUT ${link_variant}isr_tables.c
+      ${obj_copy_cmd}
+      COMMAND ${PYTHON_EXECUTABLE}
+      ${ZEPHYR_BASE}/arch/common/gen_isr_tables.py
+      --output-source ${link_variant}isr_tables.c
+      --kernel $<TARGET_FILE:${${link_variant}prebuilt}>
+      --intlist ${link_variant}isrList.bin
+      $<$<BOOL:${CONFIG_BIG_ENDIAN}>:--big-endian>
+      $<$<BOOL:${CMAKE_VERBOSE_MAKEFILE}>:--debug>
+      ${GEN_ISR_TABLE_EXTRA_ARG}
+      DEPENDS ${${link_variant}prebuilt}
+      )
+
+    set(${link_variant}generated_kernel_files ${link_variant}isr_tables.c)
+  endif()
+
   configure_linker_script(
-    linker_mcuboot_s1.cmd
-    "-DLINK_MCUBOOT_INTO_S1;-DLINKER_PASS2"
+    ${link_variant}linker.cmd
+    "-DLINK_INTO_${link_variant_name};-DLINKER_PASS2"
```

Comment:
NB: This changes the macro name to LINK_INTO_s1, which may or may not be what you want

Reply:
it is. IMO this is better than adding 19 lines of cmake to make it upper case. I will update the linker script changes.